### PR TITLE
Refactor and clean up FlashMessage component

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards giantswarm/giantswarm#12347

This PR translates the existing `FlashMessage` component and sass styles to TypeScript and styled component.

### CODE REVIEW

File: src/components/FlashMessages/FlashMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// FlashMessage
```
Comment (ubergesundheit) [inline src/components/FlashMessages/FlashMessage.tsx:1]: WDTY? Move this file into the `UI` folder?

File: src/styles/index.ts [hunk 2]
```diff
@@ -4,16 +4,19 @@ import { CSSBreakpoints } from 'shared/constants';
 
 import theme from './theme';
 
+export enum FlashMessageType {
+  danger = 'danger',
+  success = 'success',
+  warning = 'warning',
+  info = 'info',
```
Comment (axbarsan) [inline src/styles/index.ts:11]: Enum members are [usually](https://www.typescriptlang.org/docs/handbook/enums.html) PascalCased

```suggestion
  Danger = 'danger',
  Success = 'success',
  Warning = 'warning',
  Info = 'info',
```

**Review by axbarsan — COMMENTED** [review]
A few small things, otherwise looking really good 👍🏻

**Review by axbarsan — APPROVED** [review]
🚀

**Pull request merged**